progenitor-impl/src/lib.rs

```diff
@@ -581,10 +630,51 @@ pub fn validate_openapi(spec: &OpenAPI) -> Result<()> {
     Ok(())
 }
 
+/// Return all component names where schema references are keyed
+/// by multipart/form-data but nowhere else. The last part of a media
+/// schema's json path reference is considered the component name.
+fn get_exclusive_formdata_refs<'a>(
```

Comment:
these checks seem insufficient to validate if referenced types are used elsewhere e.g. what if the type is used as a body parameter or response parameter. I may be misunderstanding what's going on here, but I don't think we should do any of this.

Reply:
> what if the type is used as a body parameter or response parameter

That's what this function should find out. Alas, it's only partially validated in tests, not provably.

> what's going on here

It's to avoid that multipart/form-data body structs are emitted twice. Once with file fields factored out and once from the original ref.

> should do any of this

I thinkt so too, but haven't found a cooler way. My initial idea was way to inline types earlier, but I think the current typify API didn't let me. Maybe I'll find a better approach still.

Reply:
I would suggest we just not do anything clever here: generating orphaned types is fine--we already do it!

Reply:
Ok... good to know! Then this PR might come to a conclusion. 
It's only that without https://github.com/oxidecomputer/progenitor/pull/609/commits/f8baafaacc7d215f5c7402c4d7bce5b33c6d265f, the number of orphaned types was very, very large for our internal uses. This simply went against my sense of aesthetics. Technically, I don't think it added compile times really.

Reply:
there's probably a bigger, more holistic look one might take: start from all operations, see the types they touch. We could have special handling for these multi-part types (i.e. use in those cases wouldn't count as real use)

Reply:
Yes. OTOH, the more I thought about that, the larger the PR grew in my mind as well ;)
[Intuitively](https://www.reddit.com/r/rust/comments/17pyoo5/5_hours_to_77_seconds_how_database_tricks_sped_up/k8brg3r/), every operation could probably inferred from a generalized query into a graph of schema units.